waterutil/tun_ipv4.go

```diff
@@ -28,11 +28,42 @@ func IPv4Source(packet []byte) net.IP {
 	return net.IPv4(packet[12], packet[13], packet[14], packet[15])
 }
 
+func SetIPv4Source(packet []byte, source net.IP) {
+	copy(packet[12:16], source)
```

Comment:
(same for `SetIPv4Destination`,) Maybe use [`net.IP.To4()`](https://golang.org/pkg/net/#IP.To4) to check if `source` is indeed an IPv4 address? Or perhaps at least replace `source` with `source.To4()` so it results in a runtime error when non-IPv4 address is used?